### PR TITLE
ARTP-599 - null checks order flow

### PR DESCRIPTION
**Pull request by james-gray-adaptive**
Contains null check fixes in preparation for turning on --strictNullChecks setting

### CODE REVIEW

File: src/client/src/routes/NotificationRoute.tsx [hunk 1]
```diff
@@ -36,19 +36,16 @@ export class NotificationRoute extends React.Component<{}, State> {
...
+    return !!message ? (
       <ThemeProvider>
         <TradeNotification
           trade={message.tradeNotification}
           dismissNotification={this.onDismissNotification}
           highlightTradeInBlotter={this.highlightTradeInBlotter}
         />
       </ThemeProvider>
-    )
+    ) : null
```
Comment (Sharakai) [inline src/client/src/routes/NotificationRoute.tsx]: Wouldn't this previously always render `<TradeNotification />` but `message.tradeNotification` would be undefined? Or was that actually an erroneous scenario?

File: src/client/src/routes/OrderTicketRoute/MediaPlayer.tsx [hunk 2]
```diff
@@ -18,7 +18,7 @@ export interface State {
     position: number
     pausedAt: number
   }
-  source?: AudioBufferSourceNode
+  source?: AudioBufferSourceNode | null
```
Comment (Sharakai) [inline src/client/src/routes/OrderTicketRoute/MediaPlayer.tsx]: `AudioBufferSourceNode | null | undefined`? 😥 Any way to unify the `null | undefined`, or are they used for different scenarios?

File: src/client/src/routes/OrderTicketRoute/VoiceInput.tsx [hunk 3]
```diff
@@ -80,7 +80,10 @@ export class VoiceInput extends React.PureComponent<Props, State> {
     sessionInstance: null,
   }
 
-  static getDerivedStateFromProps({ context, source, requestSession }: Props, state: State): Partial<State> {
+  static getDerivedStateFromProps(
+    { context, source, requestSession }: Props,
+    state: State,
+  ): Partial<State> {
     let next: any = null
     const { analyser, blob, destination }: any = state
```
Comment (Sharakai) [inline src/client/src/routes/OrderTicketRoute/VoiceInput.tsx]: Not your PR, but why does this destructuring have an `any` type 🤔

File: src/client/src/routes/NotificationRoute.tsx (hunk 1 above)
Comment (Sharakai) [reply src/client/src/routes/NotificationRoute.tsx]: Ah I see now. The `TradeNotification` without a `trade` (or with the `defaultProps.trade` is.... Odd.

Comment (james-gray-adaptive) [reply src/client/src/routes/NotificationRoute.tsx]: Yeah, doesn't make sense to have a notification when all the details are not present.

File: src/client/src/routes/OrderTicketRoute/MediaPlayer.tsx (hunk 2 above)
Comment (james-gray-adaptive) [reply src/client/src/routes/OrderTicketRoute/MediaPlayer.tsx]: Good shout. Removing the undefined. Thanks :)

File: src/client/src/routes/OrderTicketRoute/VoiceInput.tsx (hunk 3 above)
Comment (james-gray-adaptive) [reply src/client/src/routes/OrderTicketRoute/VoiceInput.tsx]: Added the correct type

**Pull request merged**